controlpanel/frontend/views/parameter.py

```diff
@@ -0,0 +1,86 @@
+from django.conf import settings
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import JsonResponse
+from django.urls import reverse_lazy
+from django.views import View
+from django.views.generic import ListView, CreateView
+from django.views.generic.edit import FormMixin, DeleteView
+
+from controlpanel.api.aws import aws
+from controlpanel.api.models import Parameter
+from controlpanel.api.permissions import is_superuser
+from controlpanel.frontend.forms import CreateParameterForm
+
+
+class ParameterList(LoginRequiredMixin, ListView):
+    model = Parameter
+    template_name = "parameter-list.html"
+
+    def get_queryset(self):
+        queryset = Parameter.objects.all()
+        if is_superuser(self.request.user):
+            return queryset
+        return queryset.filter(created_by=self.request.user)
+
+
+class ParameterCreate(LoginRequiredMixin, CreateView):
+    form_class = CreateParameterForm
+    model = Parameter
+    template_name = "parameter-create.html"
+
+    def get_form_kwargs(self):
+        return FormMixin.get_form_kwargs(self)
+
+    def get_success_url(self):
+        return reverse_lazy("list-parameters")
+
+    def form_valid(self, form):
+        self.object = Parameter(
+            key=form.cleaned_data['key'],
+            role_name=form.cleaned_data['role_name'],
+            app_type=form.cleaned_data['app_type'],
+            description="",
+            created_by=self.request.user
+        )
+        self.object.value = form.cleaned_data['value']
+        self.object.save()
+        messages.success(
+            self.request,
+            f"Successfully created {self.object.name} parameter",
+        )
+        return FormMixin.form_valid(self, form)
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['app_type_choices'] = [
+            {"text": c[0], "value": c[1]}
+            for c in self.model._meta.get_field('app_type').choices
+        ]
+        return context
+
+
+
+class ParameterDelete(LoginRequiredMixin, DeleteView):
+    model = Parameter
+
+    def get_success_url(self):
+        messages.success(self.request, "Successfully delete data source")
+        return reverse_lazy("list-parameters")
+
+    def get_queryset(self):
+        queryset = Parameter.objects.all()
+        if is_superuser(self.request.user):
+            return queryset
+        return queryset.filter(created_by=self.request.user)
+
+
+class ParameterFormRoleList(LoginRequiredMixin, View):
+
+    def get(self, *args, **kwargs):
+        roles = aws.list_role_names()
+        data = {
+            "airflow": [r for r in roles if r.startswith(f"airflow")],
+            "webapp": [r for r in roles if r.startswith(f"{settings.ENV}_app")]
+        }
+        return JsonResponse(data)
```

Comment:
This view should be a DRF view in the `api` module, at a URL like `/api/cpanel/v1/parameters/`